### DISCUSSION TITLE
Add the ability to edit bookmarks

### DISCUSSION THREAD

**Discussion by dlebedinsky** [sumatrapdfreader/sumatrapdf#2480] [Ideas]
It would be amazing if you added a feature that lets users add, rename, and remove bookmarks. Favorites don't count, because they aren't saved to the PDF itself. 
I just wasted about an hour trying to add new bookmarks to my PDF without paying for Adobe Acrobat Pro. I ended up having to download an obscure, barely functional proprietary PDF editor.

**Comment by kjk** [comment]
That's unlikely to happen anytime soon.

**Comment by GitHubRulesOK** [comment]
@dlebedinsky 
Adding the outline (bookmarks) retrospectively to an existing PDF is fairly complex as it can be based on so many different file structures and user preference thus the most basic method is a list of pages with names which many editors can attempt using say the chapter heading locations but it will be different to any frontispiece ToC as the two are in no way related.

The simplest command line tool may be https://www.pdflabs.com/blog/export-and-import-pdf-bookmarks/ but it is probably the best of the low end tools and Acrobat Pro is the top end GUI for adding "Bookmarks" with complex JavaScripted actions, but there are dozens with better or more likely worse abilities.

In theory (I think some users have done so) if you use favorites to build a list of selected page entries with your desired name (so short name per target page) you can copy or parse that section of settings and use in text editor like notepad++ to restructure as you wish for input to PDFTK. The task as a result would be very basic, but reasonably customizable. the alternative is use an editor that auto builds outline  based on a text style used for headings and chapters but that can be very hit and miss, depending on sources.

**Reply by abelokoj-github** [reply]
Foxit pdf reader (the free version does bookmark editing). not sure why this is not available in Sumatra pdf.

**Comment by abelokoj-github** [comment]
I will like an option to be able to create and edit Bookmarks on pdf just like you can do in Foxit pdf reader. There is an option with the shortcut **CTRL+B** that allows you to add a page to favorites. I think there should also be a shortcut say **CRTL+SHFT+B** to add the page to bookmarks and give it a name of your choice.

Users should also be able to click on a pdf bookmark and get the option to rename, delete or change the bookmark order in any pdf document.

May thanks in anticipation. Your work and efforts is highly appreciated.

**Reply by GitHubRulesOK** [reply]
Foxit's annual revenues are $10-$50 million (see exact revenue data) and has 10-100 employees. It is classified as operating in the Software Publishers industry. I make that as a minimum 100K each so guess they must be full time employees.

**Reply by abelokoj-github** [reply]
😅😅😅.

I think I get your point now.

But it will be great to have such a feature though.
